### PR TITLE
Fixing GraphQL query as per bug #6

### LINKED ISSUE #6: Field 'license' doesn't exist on type 'Repository'
GraphQL query is wrong according to https://developer.github.com/v4/object/repository/, because the field is `licenseInfo` rather than `license`.

```
(sensibility) Luizs-MacBook-Pro:sensibility luizperes$ sensibility mine download
INFO:ConcreteLanguageProxy:Language inferred from environment: javascript
INFO:download_worker:Downloader queue: q:work:010f6c39-7b0c-4616-b5ab-5c15bdee245c
INFO:download_worker:Fetching fbeline/Design-Patterns-JS
ERROR:download_worker:Error downloading "fbeline/Design-Patterns-JS
"
Traceback (most recent call last):
  File "/Users/luizperes/Projects/sensibility/sensibility/miner/downloader.py", line 79, in loop_forever
    self.do_job(job)
  File "/Users/luizperes/Projects/sensibility/sensibility/miner/downloader.py", line 105, in do_job
    repo = self.client.fetch_repository(repo_id)
  File "/Users/luizperes/Projects/sensibility/sensibility/miner/downloader.py", line 196, in fetch_repository
    """, owner=repo.owner, name=repo.name)
  File "/Users/luizperes/Projects/sensibility/sensibility/miner/downloader.py", line 230, in query
    raise ValueError(repr(response['errors']))
ValueError: [{'message': "Field 'license' doesn't exist on type 'Repository'", 'locations': [{'line': 15, 'column': 17}]}]
INFO:download_worker:Fetching fbrandel/ParisHilton.js
ERROR:download_worker:Error downloading "fbrandel/ParisHilton.js
```

### PR DESCRIPTION
**Pull request by luizperes**
This PR fixes #6 by changing the GraphQL query as per https://developer.github.com/v4/object/repository/

### CODE REVIEW

File: sensibility/miner/downloader.py [hunk 1]
```diff
@@ -200,12 +202,15 @@ def fetch_repository(self, repo: RepositoryID) -> RepositoryMetadata:
             raise ValueError(f'Could not fetch info for {repo}')
         owner, name = RepositoryID.parse(info['nameWithOwner'])
         latest_commit = info['defaultBranchRef']['target']
+        licenseName = None
```
Comment (eddieantonio) [inline sensibility/miner/downloader.py]: Two issues here:

 1. please use `snake_case` to name variables, as per [PEP 8](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names): that is,  `license_name`
2. This variable should be declared as `str` or `Optional[str]`

File: sensibility/miner/downloader.py [hunk 2]
```diff
@@ -200,12 +202,15 @@ def fetch_repository(self, repo: RepositoryID) -> RepositoryMetadata:
...
+        if info['licenseInfo'] is not None:
+            licenseName = info['licenseInfo']['name']
```
Comment (eddieantonio) [inline sensibility/miner/downloader.py]: I'd be really helpful if you added a link to the documentation: https://developer.github.com/v4/object/license/

**Review by eddieantonio — CHANGES_REQUESTED** [review]
Thanks for addressing this! I'm glad there's interest in the project! I wrote the mining code when GitHub was giving an early opt-in preview to their GraphQL endpoint; they must have changed this since.

I can't merge this in the current state, however: this PR doesn't type-check: Please address the type error: https://travis-ci.org/naturalness/sensibility/builds/462114167#L455

File: sensibility/miner/downloader.py (hunk 2 above)
Comment (luizperes) [reply sensibility/miner/downloader.py]: you mean adding as a comment above the line `206`?

File: sensibility/miner/downloader.py (hunk 1 above)
Comment (luizperes) [reply sensibility/miner/downloader.py]: sounds good! will do.

File: sensibility/miner/downloader.py (hunk 2 above)
Comment (eddieantonio) [reply sensibility/miner/downloader.py]: Yes, please!

**Comment by luizperes** [comment]
Hi @eddieantonio, I've change most requests you made but I am a little confused about adding the url `https://developer.github.com/v4/object/license/`.

You asked me to add the documentation, but where would that be located? Is it in file `downloader.py` line `205`? I know I've asked you, I am just double-checking :)

**Comment by eddieantonio** [comment]
> You asked me to add the documentation, but where would that be located? Is it in file `downloader.py` line `205`? I know I've asked you, I am just double-checking :)

Yeah, just in a comment above line 207.

**Comment by luizperes** [comment]
will do!

File: sensibility/miner/downloader.py [hunk 3]
```diff
@@ -200,12 +202,15 @@ def fetch_repository(self, repo: RepositoryID) -> RepositoryMetadata:
             raise ValueError(f'Could not fetch info for {repo}')
         owner, name = RepositoryID.parse(info['nameWithOwner'])
         latest_commit = info['defaultBranchRef']['target']
+        license_name: Optional[str] = None
```
Comment (eddieantonio) [inline sensibility/miner/downloader.py:205]: After reading that documentation again, I realize that the License.name is declared as `String!`, meaning that info['licenseInfo']['name'] is **MUST** be a string! So you don't need to check set it as `None`, or check if it's `None`... In theory.

Comment (luizperes) [reply sensibility/miner/downloader.py:205]: Hi @eddieantonio, you are right, however, in line `214` it takes an optional string for the field `license`.

```Python
...
license=license_name,
...
```

See that `name` in the `licenseInfo` always MUST be a string!, but the field `licenseInfo` is optional, meaning that I have an optional `license_name` as it only must exist if a `licenseInfo` is __not__ `None`.

Thus: `licenseInfo <-> license_name` (and it makes the whole optional)

Comment (eddieantonio) [reply sensibility/miner/downloader.py:205]: Ah, okay. Apparently `RepositoryMetadata.license` **MUST** be a string (for reasons I can't quite remember), so you may want to change line 214 to this:

```python
license=license_name or '',
```

So if `license_name` still ends up being `None`, it is set as the empty string as a fallback. This should still type-check.

Comment (luizperes) [reply sensibility/miner/downloader.py:205]: Another thing is that I was having a crash (please see `commit dffdccded26094b44c2dfad85841fbf861e26f9b - Fixing potencial bug as the licenseInfo is an optional value`)

will make the change on line `214` you requested!

File: sensibility/miner/downloader.py [hunk 4]
```diff
@@ -200,12 +202,16 @@ def fetch_repository(self, repo: RepositoryID) -> RepositoryMetadata:
...
+        if info['licenseInfo'] is not None:
+            # as per https://developer.github.com/v4/object/license/
+            license_name = info['licenseInfo']['name']
 
         return RepositoryMetadata(
             owner=owner,
             name=name,
             revision=latest_commit['sha1'],
-            license=info['license'],
+            license=license_name or '',
```
Comment (luizperes) [inline sensibility/miner/downloader.py:214]: done as requested

**Review by eddieantonio — APPROVED** [review]

**Comment by eddieantonio** [comment]
Huzzah! It type-checks! Thanks so much for this PR!

**Pull request merged**